extensions/agent/src/dialogs/createStepDialog.ts

```diff
@@ -344,12 +360,63 @@ export class CreateStepDialog {
 		return retryFlexContainer;
 	}
 
+	private openFileBrowserDialog() {
+		let fileBrowserTitle = CreateStepDialog.FileBrowserDialogTitle + `${this.server}`;
+		this.fileBrowserDialog = sqlops.window.modelviewdialog.createDialog(fileBrowserTitle);
+		let fileBrowserTab = sqlops.window.modelviewdialog.createTab('File Browser');
+		this.fileBrowserDialog.content =  [fileBrowserTab];
+		fileBrowserTab.registerContent(async (view) => {
+			this.fileBrowserTree = view.modelBuilder.fileBrowserTree()
+				.withProperties({ ownerUri: this.ownerUri })
+				.component();
+			this.selectedPathTextBox = view.modelBuilder.inputBox()
+				.withProperties({ inputType: 'text'})
+				.component();
+			this.fileBrowserTree.onDidChange((args) => {
+				this.selectedPathTextBox.value = args.fullPath;
+				this.fileBrowserNameBox.value = args.isFile ? args.fullPath.replace(/^.*[\\\/]/, '') : '';
```

Comment:
What's this `replace` logic for? If it's something that extension authors will often need when working with the file browser component then it might be good to build it into the component as a property